src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -0,0 +1,3439 @@
+    wh_check = np.where(mask_2d_init[0, :] & ~mask_2d_init[1, :] & ~pixel_done
+                        & (end_locs == 1) & (start == 0))
+
+    if len(wh_check[0]) > 0:
+        fit_next_segment_good_0th_bad_1st(
+            wh_check, start, end_st, end_heads, got_case, ngroups)
+
+    # CASE OTHER: all other types of segments not covered earlier. No segments
+    #   handled here have adequate data, but the stack arrays are updated.
+    wh_check = np.asarray(np.where(~pixel_done & ~got_case))
+    if len(wh_check[0]) > 0:
+        fit_next_segment_all_other(wh_check, start, end_st, end_heads, ngroups)
+
+    return f_max_seg, num_seg
+
+
+def fit_next_segment_all_other(wh_check, start, end_st, end_heads, ngroups):
+    """
+    Catch all other types of segments not covered earlier. No segments
+    handled here have adequate data, but the stack arrays are updated.
+        - increment start array
+        - remove current end from end stack
+        - decrement number of ends
+
+    Parameter
+    ----------
+    wh_check: 1D array
+        pixels for current segment processing and updating
+
+    start : int, 1D array
+        lowest channel in fit
+
+    end_st : int, 2D array
+        stack array of endpoints
+
+    end_heads : int, 1D array
+        number of endpoints for each pixel
+
+    ngroups: int
+        number of groups in exposure
+    """
+    these_pix = wh_check[0]
+    start[these_pix] += 1
+    start[start > ngroups - 1] = ngroups - 1  # to keep at max level
+    end_st[end_heads[these_pix] - 1, these_pix] = 0
+    end_heads[these_pix] -= 1
+    end_heads[end_heads < 0.] = 0.
+
+
+def fit_next_segment_good_0th_bad_1st(
+        wh_check, start, end_st, end_heads, got_case, ngroups):
+    """
+    The segment has a good 0th group and a bad 1st group. For the
+    data from the 0th good group of this segment to possibly be used as a
+    slope, that group must necessarily be the 0th group of the entire ramp.
+    It is possible to have a single 'good' group segment after the 0th group
+    of the ramp; in that case the 0th group and the 1st group would both have
+    to be CRs, and the data of the 0th group would not be included as a slope.
+    For a good 0th group in a ramp followed by a bad 1st group there must be
+    good groups later in the segment because if there were not, the segment
+    would be done in `fit_next_segment_only_good_0th_group`. In this situation,
+    since here are later good groups in the segment, those later good groups
+    will be used in the slope computation, and the 0th good group will not be.
+    As a result, for all instances of these types of segments, the data in the
+    initial good group will not be used in the slope calculation, but the
+    arrays for the indices for the ramp (end_st, etc) are appropriately
+    adjusted.
+        - increment start array
+        - remove current end from end stack
+        - decrement number of ends
+
+    Parameter
```

Comment:
```suggestion
    Parameters
```

Reply:
In the docstrings, I changed "Parameter" to "Parameters".